scripts/iscsictl.py

```diff
@@ -155,10 +155,16 @@ def connect(self):
         if not self._copy_id:
             self.ssh_copy_id()
 
-        self._connect = sh.ssh.bake('-i', self.key.key(),
-                                    '-o', 'StrictHostKeyChecking=no',
-                                    '-o', 'UserKnownHostsFile=/dev/null',
-                                    '%s@%s' % (self.user, self.host))
+        if self.key:
+            self._connect = sh.ssh.bake('-i', self.key.key(),
+                                        '-o', 'StrictHostKeyChecking=no',
+                                        '-o', 'UserKnownHostsFile=/dev/null',
+                                        '%s@%s' % (self.user, self.host))
+        else:
+            self._connect = sh.ssh.bake('-o', 'StrictHostKeyChecking=no',
```

Comment:
this is code duplication

I recommend to just write the "-i" to "-o" to a variable in an if statement, then use the variable as first parameter.